Origin: upstream https://github.com/varnishcache/varnish-cache/commit/cc5df0ef06caa2fca78ba471a6126fcad03fd6a2
Description: Fix memory leak when discarding labels. This fix is included as part of 6.0.x and 6.4.x so it can be
safely removed.
--- a/bin/varnishd/cache/cache_vcl.c
+++ b/bin/varnishd/cache/cache_vcl.c
@@ -911,6 +911,8 @@ vcl_cli_discard(struct cli *cli, const c
 	if (!strcmp(vcl->state, VCL_TEMP_LABEL)) {
 		VTAILQ_REMOVE(&vcl_head, vcl, list);
 		free(vcl->loaded_name);
+		AZ(errno=pthread_rwlock_destroy(&vcl->temp_rwl));
+		FREE_OBJ(vcl);
 	} else if (vcl->temp == VCL_TEMP_COLD)
 		VCL_Poll();
 }
